dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaEventListenerQaService.kt

```diff
@@ -179,38 +179,99 @@ class QaEventListenerQaService
                 qaReviewManager.deleteAllByDataId(dataId, correlationId)
             }
         }
+
+        /**
+         * Method to retrieve message from dataStored exchange and constructing new one for qualityAssured exchange
+         * @param payload the message body as a json string
+         * @param correlationId the correlation ID of the current user process
+         * @param type the type of the message
+         */
+        @RabbitListener(
+            bindings = [
+                QueueBinding(
+                    value =
+                        Queue(
+                            QueueNames.QA_SERVICE_DATA_POINT_QA,
+                            arguments = [
+                                Argument(name = "x-dead-letter-exchange", value = ExchangeName.DEAD_LETTER),
+                                Argument(name = "x-dead-letter-routing-key", value = "deadLetterKey"),
+                                Argument(name = "defaultRequeueRejected", value = "false"),
+                            ],
+                        ),
+                    exchange = Exchange(ExchangeName.BACKEND_DATA_POINT_EVENTS, declare = "false"),
+                    key = [RoutingKeyNames.DATA_POINT_UPLOAD],
+                ),
+            ],
+        )
+        fun addDataPointToDataPointQaReviewRepository(
+            @Payload payload: String,
+            @Header(MessageHeaderKey.CORRELATION_ID) correlationId: String,
+            @Header(MessageHeaderKey.TYPE) type: String,
+        ) {
+            MessageQueueUtils.validateMessageType(type, MessageType.PUBLIC_DATA_RECEIVED)
+
+            MessageQueueUtils.rejectMessageOnException {
+                val dataUploadedPayload = MessageQueueUtils.readMessagePayload<DataUploadedPayload>(payload, objectMapper)
+                MessageQueueUtils.validateDataId(dataUploadedPayload.dataId)
+                logger.info(
+                    "Received QA required for dataId ${dataUploadedPayload.dataId} with " +
+                        "bypassQA ${dataUploadedPayload.bypassQa} (correlation Id: $correlationId)",
+                )
+
+                saveQaReviewEntityFromMessage(dataUploadedPayload, correlationId)
+            }
+        }
+
+        /**
+         * Method to save a DataPointQaReviewEntity from a QaPayload
+         * @param dataUploadedPayload the payload containing the dataId and bypassQa
+         * @param correlationId the correlation ID of the current user process
+         * @return the saved DataPointQaReviewEntity
+         */
+        fun saveQaReviewEntityFromMessage(
+            dataUploadedPayload: DataUploadedPayload,
+            correlationId: String,
+        ): DataPointQaReviewEntity {
+            val dataId = dataUploadedPayload.dataId
+            val bypassQa = dataUploadedPayload.bypassQa
+            val triggeringUserId = dataPointControllerApi.getDataPointMetaInfo(dataId).uploaderUserId ?: "No Uploader available"
```

Comment:
This is probably a bad idea: A user ID should actually be a user ID and not some random string. Let's use null if we do not have any information. In which cases would we not have any information anyway?

Reply:
Added assertion as this should never be null in the first place. Just OpenAPI stuff I guess